pkg/asset/installconfig/gcp/validation.go

```diff
@@ -141,66 +141,144 @@ func validateZoneProjects(client API, ic *types.InstallConfig, fieldPath *field.
+				errMsg := fmt.Sprintf("dns zone %s did not match expected %s", returnedZone.DnsName, ic.ClusterDomain())
+				allErrs = append(allErrs, field.Invalid(fieldPath.Child("privateManagedZone"), ic.GCP.PrivateDNSZone.ID, errMsg))
+			}
 		}
 	}
 
 	return allErrs
 }
 
-// ValidatePreExitingPublicDNS ensure no pre-existing DNS record exists in the public
-// DNS zone for cluster's Kubernetes API.
-func ValidatePreExitingPublicDNS(client API, ic *types.InstallConfig) error {
+// ValidatePreExistingPrivateDNS ensure that the PrivateZone exists in the cluster domain
+func ValidatePreExistingPrivateDNS(client API, ic *types.InstallConfig) error {
+	record := fmt.Sprintf("api.%s.", strings.TrimSuffix(ic.ClusterDomain(), "."))
+	project := internalFindProject(ic.GCP.PrivateDNSZone, ic.GCP.ProjectID)
+
+	if ic.GCP.PrivateDNSZone != nil && ic.GCP.PrivateDNSZone.ID != "" {
+		zone, err := client.GetDNSZoneByName(context.TODO(), project, ic.GCP.PrivateDNSZone.ID)
+		if err != nil {
+			return fmt.Errorf("failed to find public zone in project %s", project)
+		}
+
+		rrSets, err := client.GetRecordSets(context.TODO(), project, zone.Name)
+		if err != nil {
+			return field.InternalError(field.NewPath("baseDomain"), err)
+		}
+
+		for _, r := range rrSets {
+			if strings.EqualFold(r.Name, record) {
+				errMsg := fmt.Sprintf("record %s already exists in DNS Zone (%s/%s) and might be in use by another cluster, please remove it to continue", record, project, zone.Name)
+				return field.Invalid(field.NewPath("metadata", "name"), ic.ObjectMeta.Name, errMsg)
+			}
+		}
+	}
+	return nil
+}
+
+// ValidatePreExistingPublicDNS ensure no pre-existing DNS record exists in the public
+// DNS zone for cluster's Kubernetes API. If a PublicDNSZone is provided, the provided
+// zone is verified against the BaseDomain. If no zone is provided, the base domain is
+// checked for any public zone that can be used.
+func ValidatePreExistingPublicDNS(client API, ic *types.InstallConfig) error {
 	// If this is an internal cluster, this check is not necessary
 	if ic.Publish == types.InternalPublishingStrategy {
 		return nil
 	}
 
 	record := fmt.Sprintf("api.%s.", strings.TrimSuffix(ic.ClusterDomain(), "."))
+	project := internalFindProject(ic.GCP.PublicDNSZone, ic.GCP.ProjectID)
+	zoneName := ""
 
-	zone, err := client.GetPublicDNSZone(context.TODO(), ic.Platform.GCP.ProjectID, ic.BaseDomain)
-	if err != nil {
-		var gErr *googleapi.Error
-		if errors.As(err, &gErr) {
-			if gErr.Code == http.StatusNotFound {
-				return field.NotFound(field.NewPath("baseDomain"), fmt.Sprintf("DNS Zone (%s/%s)", ic.Platform.GCP.ProjectID, ic.BaseDomain))
-			}
+	if ic.GCP.PublicDNSZone != nil && ic.GCP.PublicDNSZone.ID != "" {
+		zoneName = ic.GCP.PublicDNSZone.ID
+	}
+
+	if zoneName == "" {
+		zone, err := client.GetPublicDNSZone(context.TODO(), project, ic.BaseDomain)
+		if err != nil {
+			return fmt.Errorf("failed to find public zone in project %s", project)
```

Comment:
can you `wrap` `err` with this message, so that err gets passed through to stdout